src/GridFS/CollectionWrapper.php

```diff
@@ -150,14 +152,17 @@ public function findFileByFilenameAndRevision(string $filename, int $revision):
             $sortOrder = 1;
         }
 
-        return $this->filesCollection->findOne(
+        $file = $this->filesCollection->findOne(
             ['filename' => $filename],
             [
                 'skip' => $skip,
                 'sort' => ['uploadDate' => $sortOrder],
                 'typeMap' => ['root' => 'stdClass'],
             ]
         );
+        assert(is_object($file) || $file === null);
```

Comment:
I assume this is another case where psalm can't infer how the typeMap will be honored, correct?

Reply:
That is correct, yes.